src/js/base/module/Editor.js

```diff
@@ -204,7 +205,7 @@ export default class Editor {
 
       if (this.options.onCreateLink) {
         linkUrl = this.options.onCreateLink(linkUrl);
-      } else {
+      } else if (checkProtocol) {
         // if url doesn't have any protocol and not even a relative or a label, use http:// as default
         linkUrl = /^([A-Za-z][A-Za-z0-9+-.]*\:|#|\/)/.test(linkUrl)
           ? linkUrl : 'http://' + linkUrl;
```

Comment:
Looks good. It would be better if we can pull out the `http://` into `settings.js` as something like  `defaultLinkProtocol`.